File: engine/transpile/src/test/resources/golden/SYK008/WS_応答コード.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK008 のデータ部レコード WS-応答コード を逐語対訳した自動生成コード。 */
public final class WS_応答コード {

    /** 01 WS-応答コード 総バイト長 8。 */
    public static final int LENGTH = 8;

    private final byte[] data;

    public WS_応答コード() {
        this.data = new byte[8];
    }

    public WS_応答コード(byte[] source) {
        this.data = Arrays.copyOf(source, 8);
    }

    public byte[] data() {
        return data;
    }

    // WS-RESPコード
    public long get_WS_RESPコード() {
        return CobolRuntime.decodeBinary(data, 0, 4, true);
    }
    public void set_WS_RESPコード(long value) {
        CobolRuntime.encodeBinary(data, 0, 4, true, value);
    }

    // WS-RESP2コード
    public long get_WS_RESP2コード() {
        return CobolRuntime.decodeBinary(data, 4, 4, true);
    }
    public void set_WS_RESP2コード(long value) {
        CobolRuntime.encodeBinary(data, 4, 4, true, value);
    }
}

File: engine/transpile/src/test/resources/golden/SYK008/WS_応答コード.py
"""SYK008 のデータ部レコード WS-応答コード を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_応答コード:
    """01 WS-応答コード 総バイト長 8。"""

    LENGTH = 8

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(8)
        else:
            self._data = bytearray(data)
            if len(self._data) < 8:
                self._data.extend(b"\x00" * (8 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-RESPコード
    def get_WS_RESPコード(self):
        return decode_binary(self._data, 0, 4, True)
    def set_WS_RESPコード(self, value):
        encode_binary(self._data, 0, 4, True, value)

    # WS-RESP2コード
    def get_WS_RESP2コード(self):
        return decode_binary(self._data, 4, 4, True)
    def set_WS_RESP2コード(self, value):
        encode_binary(self._data, 4, 4, True, value)
